services/wallet/service.go

```diff
@@ -59,7 +60,11 @@ func NewService(
 		Publisher: walletFeed,
 	}
 	blockchainStatus := make(map[uint64]string)
+	mutex := sync.Mutex{}
 	rpcClient.SetWalletNotifier(func(chainID uint64, message string) {
```

Comment:
The best is to move this listener code outside the init function to have a cleaner implementation and make mutex a member of the service and call it `rpcClientMutex`